src/Arcus.Scripting.ApiManagement/Scripts/Backup-AzApiManagementService.ps1

```diff
@@ -23,32 +29,64 @@ if ($storageKeys -eq $null -or $storageKeys.count -eq 0) {
     Write-Host "New Azure storage context for storage account '$($StorageAccountName)' with storage key created!" -ForegroundColor Green
 
     Write-Verbose "Start backing up Azure API Management instance '$($ServiceName)' in resource group '$($ResourceGroupName)'..."
-    if ($BlobName -ne $null) {
+    if ($BlobName -ne $null -and $BlobName -ne "") {
         if ($PassThru) {
             if ($DefaultProfile -ne $null) {
-                Backup-AzApiManagement -ResourceGroupName $ResourceGroupName -Name $ServiceName -StorageContext $storageContext -TargetContainerName $ContainerName -TargetBlobName $BlobName -PassThru -DefaultProfile $DefaultProfile
+                if ($AccessType -eq 'UserAssignedManagedIdentity') {
+                    Backup-AzApiManagement -ResourceGroupName $ResourceGroupName -Name $ServiceName -AccessType $AccessType -IdentityClientId $IdentityClientId -StorageContext $storageContext -TargetContainerName $ContainerName -TargetBlobName $BlobName -PassThru -DefaultProfile $DefaultProfile
+                } else {
+                    Backup-AzApiManagement -ResourceGroupName $ResourceGroupName -Name $ServiceName -AccessType $AccessType -StorageContext $storageContext -TargetContainerName $ContainerName -TargetBlobName $BlobName -PassThru -DefaultProfile $DefaultProfile
+                }
             } else {
-                Backup-AzApiManagement -ResourceGroupName $ResourceGroupName -Name $ServiceName -StorageContext $storageContext -TargetContainerName $ContainerName -TargetBlobName $BlobName -PassThru
+                if ($AccessType -eq 'UserAssignedManagedIdentity') {
+                    Backup-AzApiManagement -ResourceGroupName $ResourceGroupName -Name $ServiceName -AccessType $AccessType -IdentityClientId $IdentityClientId -StorageContext $storageContext -TargetContainerName $ContainerName -TargetBlobName $BlobName -PassThru
+                } else {
+                    Backup-AzApiManagement -ResourceGroupName $ResourceGroupName -Name $ServiceName -AccessType $AccessType -StorageContext $storageContext -TargetContainerName $ContainerName -TargetBlobName $BlobName -PassThru
+                }
             }
         } else {
             if ($DefaultProfile -ne $null) {
```

Comment:
I think that this was implemented a bit harder than need be. I believe that we can pass `$DefaultProfile = $null` to the `Bckup-AzApiMnagement`. Best that we do some refactoring here to make it more simple instead of 4 nested `if`'s.

Reply:
Changed in [`356a092` (#427)](https://github.com/arcus-azure/arcus.scripting/pull/427/commits/356a092f0ef1b33522e67922ca6ad4724f09dd83)

Reply:
It's still a lot of `if`'s 😜 Can't we use like PowerShell's argument-collection feature to pass the arguments instead of defining them ourselves? Might simplify things. https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_splatting?view=powershell-7.4

Reply:
Good one, updated the script, still some if statements but quite a bit simpler I think: [`a6dad91` (#427)](https://github.com/arcus-azure/arcus.scripting/pull/427/commits/a6dad91c5a4561b95293114c89ad2bcd9e33c636)

Reply:
Yes, like this very much, much more reaable. Thanks a lot!!